lib/themed.tcl

```diff
@@ -4,46 +4,152 @@
 
 namespace eval color {
 	# Variable colors
-	# Preffered way to set widget colors is using add_option.
-	# In some cases, like with tags in_diff/in_sel, we use these colors.
-	variable select_bg				lightgray
-	variable select_fg				black
-	variable inactive_select_bg		lightgray
-	variable inactive_select_fg		black
+	# Preferred way to set widget colors is using add_option.
+	# These variables are used by set_selection_colors.
+	variable select_bg              #242470
+	variable select_fg              #f0f0f0
+	variable inactive_select_bg     #404080
+	variable inactive_select_fg     #f0f0f0
 
 	proc sync_with_theme {} {
-		set base_bg				[ttk::style lookup . -background]
-		set base_fg				[ttk::style lookup . -foreground]
-		set text_bg				[ttk::style lookup Treeview -background]
-		set text_fg				[ttk::style lookup Treeview -foreground]
-		set select_bg			[ttk::style lookup Default -selectbackground]
-		set select_fg			[ttk::style lookup Default -selectforeground]
-		set inactive_select_bg	[convert_rgb_to_gray $select_bg]
-		set inactive_select_fg	$select_fg
-
-		set color::select_bg $select_bg
-		set color::select_fg $select_fg
-		set color::inactive_select_bg $inactive_select_bg
-		set color::inactive_select_fg $inactive_select_fg
+		# TODO: only override these if a dark-mode setting is in effect.
+
+		set darkest                 #101010
+		set darker                  #242424
+		set dark                    #303030
+		set darkish                 #404040
+		set lightish                #484848
+		set light                   #585858
+		set lighter                 #686868
+		set lightest                #808080
+		set white                   #e0e0e0
+		set whiter                  #f0f0f0
+		set whitest                 #ffffff
+
+		set scale                   18
```

Comment:
Feel free to adjust these for accessibility. I'll probably add "high contrast" and "larger controls" settings.